services/dedup/dedup.go

```diff
@@ -17,9 +17,18 @@ import (
 	"github.com/rudderlabs/rudder-server/utils/misc"
 )
 
-type DedupI interface {
-	FindDuplicates(messageIDs []string, allMessageIDsSet map[string]struct{}) (duplicateIndexes []int)
-	MarkProcessed(messageIDs []string) error
+type Info struct {
+	MessageID string
+	Size      int
+}
+
+type Payload struct {
+	Size int `json:"size"`
+}
+
+type Dedup interface {
+	FindDuplicates(infos []Info, allMessageIDsSet map[string]Payload) map[int]Payload
```

Comment:
The method didn't and still doesn't appear provide a clear enough API with respect to  Dedup's responsibilities. More specifically, `allMessageIDsSet` seems a bit out of place. The caller should be responsible for performing this kind of deduplication (along with the deduplication of the infos slice itself), since it has nothing to do with state persisted in dedup's storage itself. The fact that we don't include the payload size in the aforementioned dedup scenarios is indicative of this api design weakness.

Reply:
Yeah, this will simplify the dedup logic for the processor. 😄